plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,355 @@
+			fields[key] = values[i]
+		}
+		m := metric.New(measurement, tags, fields, t)
+		return m, nil
+	case "bool8",
+		"int8", "uint8",
+		"int16", "uint16",
+		"int32", "uint32",
+		"int64", "uint64",
+		"float", "double",
+		"string",
+		"timestamp":
+		fields := map[string]interface{}{fieldName: em.Value}
+		m := metric.New(measurement, tags, fields, t)
+		return m, nil
+	}
+
+	return nil, fmt.Errorf("unsupported value type: %s", em.Type)
+}
+
+// Init is for setup, and validating config
+func (s *CtrlXDataLayerSSE) Init() error {
+	// Check all configured subscriptions for valid settings
+	for i := range s.Subscription {
+		sub := &s.Subscription[i]
+		sub.applyDefaultSettings()
+		err := choice.Check(sub.QueueBehaviour, queueBehaviours)
+		if err != nil {
+			s.Log.Infof("The right queue behaviour values are %v", queueBehaviours)
+			return fmt.Errorf("subscription %d: setting 'queue_behaviour' %q is invalid", i, sub.QueueBehaviour)
+		}
+		err = choice.Check(sub.ValueChange, valueChanges)
+		if err != nil {
+			s.Log.Infof("The right value change values are %v", valueChanges)
+			return fmt.Errorf("subscription %d: setting 'value_change' %q is invalid", i, sub.ValueChange)
+		}
+		if len(sub.Nodes) == 0 {
+			s.Log.Warn("A configured subscription has no nodes configured")
+		}
+		sub.index = i
+	}
+
+	// Generate valid communication url based on configured hostname
+	u := url.URL{
+		Scheme: "https",
+		Host:   s.Hostname,
+	}
+	s.url = u.String()
+	_, err := url.Parse(s.url)
+	if err != nil {
+		return errors.New("invalid hostname")
+	}
+
+	return nil
+}
+
+// Start input as service, retain the accumulator, establish the connection.
+func (s *CtrlXDataLayerSSE) Start(acc telegraf.Accumulator) error {
+	var ctx context.Context
+	ctx, s.cancel = context.WithCancel(context.Background())
+
+	var err error
+	s.connection, err = s.HTTPClientConfig.CreateClient(ctx, s.Log)
+	if err != nil {
+		return fmt.Errorf("failed to create http client, err: %w", err)
+	}
+
+	s.tokenManager = token.TokenManager{
+		Url:        s.url,
+		Username:   s.Username,
+		Password:   s.Password,
+		Connection: s.connection,
+	}
+
+	s.acc = acc
+
+	s.gatherLoop(ctx)
+
+	return nil
+}
+
+// gatherLoop creates sse subscriptions on the Data Layer and requests the sse data
+// the connection will be restablished if the sse subscription is broken.
+func (s *CtrlXDataLayerSSE) gatherLoop(ctx context.Context) {
+	for _, sub := range s.Subscription {
+		s.wg.Add(1)
+		go func(sub Subscription) {
+			defer s.wg.Done()
+			for {
+				select {
+				case <-ctx.Done():
+					s.Log.Debugf("Gather loop for subscription %d stopped", sub.index)
+					return
+				default:
+					client, err := s.createSubscriptionAndSseClient(&sub)
+					if err != nil {
+						s.Log.Errorf("Creating sse client to subscription %d, %v", sub.index, err)
+						time.Sleep(time.Duration(defaultReconnectInterval))
+						continue
+					}
+					s.Log.Debugf("Created sse client to subscription %d", sub.index)
```

Comment:
Do you really have to recreate the client in every cycle? Can't we reuse the client? If not, please add a comment to help the reader understand your needs.